src/Sarif.Viewer.VisualStudio.Shell.Core/GitExe.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.Sarif.Viewer.Shell
+{
+    public class GitExe : IGitExe
+    {
+        private readonly IServiceProvider serviceProvider;
+
+        // private string repoPath;
+        private string vsInstallDir;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GitExe"/> class.
+        /// </summary>
+        /// <param name="serviceProvider">The service provider.</param>
+        public GitExe(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        /// <inheritdoc cref="IGitExe.RepoPath"/>
+        public string RepoPath { get; set; }
+
+        /// <inheritdoc cref="IGitExe.GetRepoRootAsync"/>
+        public async ValueTask<string> GetRepoRootAsync() // TODO: <string?>
+        {
+            return await ExecuteGitCommandAsync("rev-parse --show-toplevel");
+        }
+
+        /// <inheritdoc cref="IGitExe.GetRepoUriAsync"/>
+        public async ValueTask<string> GetRepoUriAsync() // TODO: <string?>
+        {
+            return await ExecuteGitCommandAsync("config --get remote.origin.url");
+        }
+
+        /// <inheritdoc cref="IGitExe.GetCurrentBranchAsync"/>
+        public async ValueTask<string> GetCurrentBranchAsync() // TODO: <string?>
+        {
+            return await ExecuteGitCommandAsync("symbolic-ref --short HEAD");
+        }
+
+        /// <inheritdoc cref="IGitExe.GetCurrentCommitHashAsync"/>
+        public async ValueTask<string> GetCurrentCommitHashAsync() // TODO: <string?>
+        {
+            return await ExecuteGitCommandAsync("rev-parse HEAD");
+        }
+
+        private async ValueTask<string> ExecuteGitCommandAsync(string arguments)
+        {
+            if (string.IsNullOrWhiteSpace(this.vsInstallDir))
+            {
+                this.vsInstallDir = await GetVsInstallDirectoryAsync();
+            }
+
+            // Get the trusted min Git executable path.
+            string minGitPath = Path.Combine(this.vsInstallDir, @"CommonExtensions\Microsoft\TeamFoundation\Team Explorer\Git\cmd\git.exe");
```

Comment:
>Path.Combine(this.vsInstallDir, @"CommonExtensions\Microsoft\TeamFoundation\Team Explorer\Git\cmd\git.exe") [](http://example.com/codeflow?start=32&length=107)

if user install git separately using official git website and does not install it with the VS installer will this have problem

Reply:
fyi I see there is a `git --exec-path` command can find where current working git location

Reply:
Yes you're right. I have several installs on my machine but I haven't found a reliable way to get the path. I'll see if I can run it by just using "git.exe".

Reply:
that works. I'll do some more digging to make sure it's reliable